src/menus/menu.jsx

```diff
@@ -329,55 +341,52 @@ const Menu = React.createClass({
     let menuItemHeight = desktop ? 32 : 48;
 
     //MaxHeight isn't set - cascade all of the children
-    if (!maxHeight) return React.Children.count(children);
+    if (!maxHeight) return filteredChildren.length;
 
     //Count all the children that will fit inside the
     //max menu height
-    React.Children.forEach(children, (child) => {
+    for (let child of filteredChildren) {
       if (currentHeight < maxHeight) {
         let childIsADivider = child.type && child.type.displayName === 'MenuDivider';
 
         currentHeight += childIsADivider ? 16 : menuItemHeight;
         count++;
       }
-    });
+    }
 
     return count;
-
   },
 
-  _getMenuItemCount() {
+  _getMenuItemCount(filteredChildren) {
     let menuItemCount = 0;
-    React.Children.forEach(this.props.children, (child) => {
+    for (let child of filteredChildren) {
```

Comment:
In my output I see the transformation, but let's not take any chances. Alright I'll Fix this. Thanks for the tip :+1: :+1: